ui/src/app/base/components/PowerTypeFields/PowerTypeFields.tsx

```diff
@@ -101,12 +117,30 @@ export const PowerTypeFields = <F extends Record<string, unknown>>({
               value: powerType.name,
             })),
           ]}
+          onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
```

Comment:
I think what this does is loop through and reset any fields that are for the selected power type. Should it also reset the values for the old power type (or maybe just everything) as couldn't it leave values in the formik state for the old fields? Not sure exactly how that works.

Either way, it would probably also good to leave a comment here explaining why this is necessary.

Reply:
It shouldn't be necessary to reset any other fields than from the selected power type because the power parameters are trimmed before the websocket call is made. Resetting everything won't work (at least using `resetForm()`) because it'll end up resetting non-power-related form fields as well.

I'll leave a comment explaining what this is for (and add a test)